File: examples-sdk/python/305-rag-guardrail-pipeline.py
#!/usr/bin/env python3
"""305-rag-guardrail-pipeline — Python SDK equivalent
Compile: vil compile --from python --input 305-rag-guardrail-pipeline.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("rag-guardrail-pipeline", port=3114)
rag_guardrail = server.service_process("rag-guardrail")
rag_guardrail.endpoint("POST", "/safe-rag", "safe_rag_handler")
server.compile()

File: examples-sdk/typescript/305-rag-guardrail-pipeline.ts
#!/usr/bin/env tsx
// 305-rag-guardrail-pipeline — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 305-rag-guardrail-pipeline.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("rag-guardrail-pipeline", 3114);
const rag_guardrail = new ServiceProcess("rag-guardrail");
rag_guardrail.endpoint("POST", "/safe-rag", "safe_rag_handler");
server.service(rag_guardrail);
server.compile();
